src/DebugChannel.js

```diff
@@ -42,14 +44,36 @@ export class DebugChannel {
   /* Request object based on the protocol
   */
   readIn(): string {
-    let contents = "";
-    while (contents.length === 0) {
-      contents = this.fs.readFileSync(this.inFilePath, "utf8");
+    let message = "";
+    while (true) {
+      let contents = this.fs.readFileSync(this.inFilePath, "utf8");
+
+      if (contents.length === 0) {
+        continue;
+      }
+      // format: <length>--<contents>
+      let separatorIndex = contents.indexOf(TWO_DASH);
+      if (separatorIndex === -1) {
+        continue;
+      }
+
+      let messageLength = parseInt(contents.slice(0, separatorIndex), 10);
+      if (isNaN(messageLength)) {
+        continue;
```

Comment:
When will this happen? Isn't this a violation of protocol and should be an invariant or throw?